pkg/controller/storagecluster/reconcile.go

```diff
@@ -1037,3 +1038,96 @@ func getMonCount(nodeCount int) int {
 
 	return count
 }
+
+func newCleanupJob(sc *ocsv1.StorageCluster, RookImage string, OsdID int, reqLogger logr.Logger) *batchv1.Job {
```

Comment:
Remaining work items:
- Wrap the Job in an OpenShift template.
- Call this method from the reconcile loop so it is kept up-to-date. 
- When there is an upgrade we need to ensure the template is updated. For example, in the 4.5 release this template will likely change once we have a simplified rook job available.